### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by hunter4466**
Interactions

- Splash screen

> A simple page with the name of my app and links to the signup and login pages.

- Sign up and login pages

> The user is able to register in the app with their full name, email, and password.
> The user can log into the app using email and password.
> If the user is not logged in, they can't access pages that require the user to be logged in.

- Home page (categories page)

> When the user logs in, they are presented with the categories page.
> For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
> When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
> There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page

> For a given category, the list of transactions is presented, ordered by the most recent.
> At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
> There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
> When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

> The user fills out a form to create a new category, indicating their name and icon.
> The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
> When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

> The user fills out a form to create a new transaction with:

- > name.
- > amount.
- > categories.

> The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
> When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

**Review by o-rashad — COMMENTED** [review]
Please see the above review :arrow_down:

**Review by o-rashad — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Asslam Alikom @hunter4466  👋😄,

I'm Omar and I will be your reviewer for this project 😊
It has been long a journey for you..
 >Successful people do what unsuccessful people are not willing to do. Don't wish it were easier; wish you were better.
>-Jim Rohn


Good job so far, the amount of efforts you put in this project is tremendous !👌😃
 - Great video with a code explanation :100: 
 - Pull request description is perfect showing all the project features :+1:

There are some issues 🐞 that you still need to work on to go to the next project but you are almost there! 🏁🏁🏁

## Required Changes ♻️
_Check the comments under the review._

1. Documentation:
   - - [ ] The `README` should contains an `Acknowledgements` section or similar that gives appropriate credit to [Gregoire Vella](https://www.behance.net/gregoirevella) on Behance the author of the [original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license](https://creativecommons.org/licenses/by-nc/4.0/) of the design. 
   - - [ ] These Steps should be added to your `README` ,so nontechnical people can work with your project. :nerd_face: 
     -  `rails db:create`
     - `  rails db:migrate`
1. Live version: 
   - - [ ] Currently the live version is not working when i visit `/users/sign_in` or `users/sign_up`, kindly fix it.

1. Interactions :
   - - [ ] Currently i can't login to the app giving me an error of i need to confirm my email first, and nothing has sent to my email, please fix the issue
        - > I tried to login with the credentials that came with the database but it didn't work also giving an error of `Invalid Email or password` 
   - - [ ] Not able to view `Home page (catagories page)`
   - - [ ] Not able to view `Transactions page`
   - - [ ] Not able to view `"Add a new category" page`
   - - [ ] Not able to view `"Add a new transaction" page`

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please Fix theses issues and ask for another review

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/categories_page_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      fill_in 'Password', with: '111111'
+      click_button 'Log in'
+      click_link 'ADD NEW CATEGORY'
+      fill_in 'Name', with: 'new category'
+      fill_in 'Place your icon url here', with: 'https://lorempixel.com/200/200/food/'
+      click_button 'CREATE'
+      expect(page).to have_content('new category')
+    end
+  end
+end
```
Comment (Emmyn5600) [inline spec/categories_page_spec.rb:72]: - [ ] Optional suggestion

You don't need to visit URL more than once in the test. You can visit the URL once and click the link or buttons.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi, @hunter4466 

Good job so far! 👏🏼 

Great work on making the project requirements:clap:
You've done well implementing the project requirements, but there are still some issues which aren't been addressed yet 
 
## STATUS: CHANGES REQUESTED :recycle:

### Changes requested.

- [ ] Live version:

As requested by the previous reviewer,  Currently, the live version is not working when I visit /users/sign_in or users/sign_up, kindly fix it.

 #### Optional suggestions
 
 - Check the comment under the review.
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**